app/controllers/plans_controller.rb

```diff
@@ -174,7 +174,11 @@ def share
     @plan = Plan.find(params[:id])
     authorize @plan
     @visibility = @plan.visibility.present? ? @plan.visibility.to_s : Rails.application.config.default_plan_visibility
-    @allow_visibility = (@plan.num_answered_questions >= 1 && !@plan.is_test?)
+
+    nanswers = @plan.num_answered_questions()
+    nquestions = @plan.num_questions()
+    value=(nanswers.to_f/nquestions*100).round(2)
+    @allow_visibility = (value >= 50.00 && !@plan.is_test?)
```

Comment:
Extracting 50 to a constant will make the code more reusable. The visibility restriction of 50% answered might not apply to any running instance.